query-engine/core/src/query_ast/write.rs

```diff
@@ -222,67 +251,60 @@ impl CreateManyRecords {
 #[derive(Debug, Clone)]
 #[allow(clippy::enum_variant_names)]
 pub enum UpdateRecord {
-    /// Update with explicitly selected fields that will eventually be serialized as results.
-    WithExplicitSelection(UpdateRecordWithSelection),
-    /// Update with implicit selection set (primary_identifier) that will only be used to fulfill other nodes requirements.
-    WithImplicitSelection(UpdateRecordWithoutSelection),
+    /// Update with explicitly selected fields.
+    WithSelection(UpdateRecordWithSelection),
     /// Update without any selection set. A subsequent read is required to fulfill other nodes requirements.
     WithoutSelection(UpdateRecordWithoutSelection),
 }
```

Comment:
`UpdateRecord::WithImplicitSelection` was removed. I did that because the field selection update required me to mutate a `UpdateRecord::WithImplicitSelection` to a `UpdateRecord::WithExplicitSelection` to specify the new selection set, which required a `name` field that a `UpdateRecord::WithImplicitSelection` didn't have. So the abstraction wasn't holding well. We now only have `UpdateRecord::WithSelection` and `UpdateRecord::WithoutSelection`, where `UpdateRecord::WithSelection` has an optional `name: Option<String>`. When a name is present, we consider that the result will eventually be serialized and so the connector's interface `fn update_record` method returns a `QueryResult::RecordSelection` instead of a `QueryResult::Id`.